lib/tempfile_factory.rb

```diff
@@ -0,0 +1,25 @@
+# Middleware to override the default Rack::Multipart tempfile factory: 
+# https://www.rubydoc.info/gems/rack/Rack/Multipart/Parser#TEMPFILE_FACTORY-constant
+class TempfileFactory
```

Comment:
I think it would be good if you could add some tests for this class if it isn't too hard.